Dockerfile

```diff
@@ -1,5 +1,4 @@
-# used JDK11 because on jdk-8 I get SUREFIRE-1588. Debian Docker images are a bit dead to me :(
-FROM maven:3.6.0-jdk-11
+FROM maven:eclipse-temurin
```

Comment:
Both Maven and JDK version must be pinned to ensure reproducibility of the build over time.

But it is 👍  to bump to latest Temurin JDK21 with Maven 3.9.10.

I suggest using 2 `ARG` instructions (example with the same kind of usage in https://github.com/jenkins-infra/docker-packaging/blob/eed87c8f7bfb414d97490c4888056a96d3bbcc5c/Dockerfile#L1-L7) so we can implement version tracking in a subsequent PR with `updatecli`.